micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/CaffeineCacheMetrics.java

```diff
@@ -121,50 +122,75 @@ public CaffeineCacheMetrics(Cache<?, ?> cache, String cacheName, Iterable<Tag> t
 
     @Override
     protected Long size() {
-        return cache.estimatedSize();
+        Cache<?, ?> ref = cache.get();
```

Comment:
An improvement for these could be a common method that receives:
- a lambda that returns a value based on the cache
- a default value in case the reference is null

Reply:
Good idea :+1: 

Two different common methods, one for `Long` and one for `long`, or what's your preference?

Reply:
Pushed a version with two methods now for `CaffeineCacheMetrics`, `GuavaCacheMetrics` and `EhCache2Metrics`.

Reply:
I think what you did is great. It's not a must at the moment but I think the method with long can be more generic by doing something like this: `private <T> T getOrDefault(Function<Cache<?, ?>, T> function, @Nullable T defaultValue)`.

I don't think this is very pressing since we only have one non-primitive type now but could be a nice touch for the future.